### PR TITLE
Update golangci-lint

### PR DESCRIPTION
**Pull request by matejvasek**
Update golangci-lint.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -17,6 +18,40 @@ linters:
...
+    - whitespace
+    - unparam
+    - gofmt
+    - gofumpt
+    - goimports
+    - gci
+    - asciicheck
+    - godot
+    - makezero
+    - unused
```
Comment (matejvasek) [inline .golangci.yml:54]: These seems to crash linter, the about are just failing.

Comment (matejvasek) [inline .golangci.yml:54]: These seems to crash linter, the about are just failing.

File: pkg/domain/infra/abi/manifest.go [hunk 2]
```diff
@@ -1,5 +1,3 @@
-// +build !remote
```
Comment (matejvasek) [inline pkg/domain/infra/abi/manifest.go:1]: Without removing this the `tunnel` linting will fail.

**Comment by matejvasek** [comment]
This update should lower memory requirements for linter. With the older version I saw 8.9GiB memory consumption. With the newer just 1.5GB.

File: .golangci.yml [hunk 3]
```diff
@@ -17,6 +18,38 @@ linters:
   enable-all: true
   disable:
     # All these break for one reason or another
+    - nolintlint
```
Comment (matejvasek) [inline .golangci.yml:21]: I think that overtime we want to actually fix some of those errors instead of suppressing them.

Comment (matejvasek) [reply .golangci.yml:21]: I don't not mean `nolintlint` specifically here, mean all suppressed checkers.

File: .golangci.yml [hunk 4]
```diff
@@ -17,6 +18,38 @@ linters:
...
+    - predeclared
+    - thelper
+    - ifshort
+    - staticcheck
+    - forbidigo
+    - govet
+    - exhaustive
+    - whitespace
+    - unparam
+    - gofmt
```
Comment (vrothberg) [inline .golangci.yml:47]: The list looks way to big to me. `gofmt` is a standard linter. If that doesn't work, we should really fix the underlying issue.

Comment (matejvasek) [reply .golangci.yml]: I enabled `gofmt` `goimports` and `govet`.

Comment (vrothberg) [reply .golangci.yml]: Thanks a lot!

**Review by vrothberg — COMMENTED** [review]
LGTM, thanks!

Now we have a good base to start enabling more linters.

**Comment by matejvasek** [comment]
One thing I don't understand: event without this PR there is a lint error:
```
conmon/cmd/conmon-config/conmon-config.go:33:13: unusedresult: result of fmt.Errorf call not used (govet)
                fmt.Errorf(err.Error())
```
however CI is not bothered by it.

**Comment by matejvasek** [comment]
@vrothberg flake?

**Comment by vrothberg** [comment]
> @vrothberg flake?

Restarted :) Looks like one

**Comment by vrothberg** [comment]
@containers/podman-maintainers PTAL

File: pkg/domain/infra/abi/manifest.go (hunk 2 above)
Comment (matejvasek) [reply pkg/domain/infra/abi/manifest.go:1]: @rhatdan is this ok?

**Review by saschagrunert — APPROVED** [review]
LGTM

**Pull request merged**